### DISCUSSION TITLE
how to use render property of STColumn like ant design ?

### DISCUSSION THREAD

**Discussion by mouad-mahfoud** [ng-alain/ng-alain#2366] [Q&A]
i have this column 
```ts
{
      title: this.i18n.fanyi('form.label.statut'),
      filter: {
        type: 'keyword',
        fn: ({ value }, record) => {
          return value ? isIncludeString(record, 'statut', value) : true;
        }
      },
      type: 'widget',
      widget: { type: 'text', params: ({ record }) => ({ text: this.i18n.fanyi(StatutPlanAction[record.statut - 1].value) }) }
    }
```

I try to use the widget to customize the rendering of column, but it doesn't work :/

That's one example when I use ant design and render property.

```ts
{
    title: <Trans>app.columns.statut</Trans>,
    dataIndex: 'statut',
    key: 'statut',
    align: 'center',
    render: (_, record) => <Trans>{statusList[record?.statut - 1]?.value}</Trans>
  },
```

**Comment by cipchk** [comment]
Can you provider a reproduction demo via https://stackblitz.com/edit/ng-alain-setup?file=src%2Fapp%2Fapp.component.ts

**Reply by mouad-mahfoud** [reply]
hello @cipchk ,

here is the demo : https://stackblitz.com/edit/ng-alain-setup-h2vzt4?file=src%2Fapp%2Fapp.component.ts
check please the line 171 in app.component.ts

thanks .

**Reply by cipchk** [reply]
I think what you need is `format` not `render`, Of course both ways are fine, like this: https://stackblitz.com/edit/ng-alain-setup-v7kkyh?file=src%2Fapp%2Fapp.component.ts

**Reply by mouad-mahfoud** [reply]
It works for me, thank's @cipchk 👍
